pkg/provisioners/channel_util.go

```diff
@@ -57,16 +62,22 @@ func AddFinalizer(o metav1.Object, finalizerName string) AddFinalizerResult {
 	return FinalizerAdded
 }
 
-func RemoveFinalizer(o metav1.Object, finalizerName string) {
+func RemoveFinalizer(o metav1.Object, finalizerName string) RemoveFinalizerResult {
+	result := FinalizerNotFound
 	finalizers := sets.NewString(o.GetFinalizers()...)
-	finalizers.Delete(finalizerName)
-	o.SetFinalizers(finalizers.List())
+	if finalizers.Has(finalizerName) {
```

Comment:
My preference would be:

```golang
	if finalizers.Has(finalizerName) {
		finalizers.Delete(finalizerName)
		o.SetFinalizers(finalizers.List())
		return FinalizerRemoved
	}
	return FinalizerNotFound
```

Reply:
Will make this change in the next PR which will include kafka related changes.